lib/util.ex

```diff
@@ -67,27 +73,33 @@ defmodule Util do
       Authorization: "Bearer " <> auth
     ]
 
-    {:ok, resp} = HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers)
-
-    resp.body
-    |> JSON.decode!()
-    |> Map.get("data")
-    |> Map.get("tournaments")
-    |> Map.get("nodes")
-    # Finesse data into our own non-start.gg format.
-    |> Enum.map(fn sgg_data ->
-      %{
-        "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-        "name" => sgg_data["name"],
-        "location" => %{
-          "lat" => sgg_data["lat"],
-          "lng" => sgg_data["lng"]
-        },
-        "start_time" => sgg_data["startAt"],
-        "end_time" => sgg_data["endAt"],
-        "updated_at" => sgg_data["updatedAt"],
-        "url" => "https://start.gg/" <> (sgg_data["slug"])
-      }
-    end)
+    case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
+      {:ok, resp} ->
+        case JSON.decode(resp.body) do
+          {:ok, response = %{"success" => false}} ->
+            IO.puts(Map.get(response, "message"))
```

Comment:
Cleaned things up a bit and just pattern matched on the JSON.decode response, the main error I can force is just having an incorrect auth token, which here will reliably print `Invalid authentication token`.

The issue that I'm trying to handle now is that since in the `get_all_tourneys` call further up, we run an Enum.map 10 times regardless of if an error gets IO.puts out or not, so we get 10 `Invalid authentication token`, which will presumably also happen for other errors that could occur. I'm wondering if have any thoughts on handling that, and also if we want to handle any downstream actual BadMapErrors from the `Map.get` calls we make once we get back a presumably successful response back.

Reply:
re: repeated errors, I responded in the comment above. Let's fix that in a separate PR, see issue #58.

> if we want to handle any downstream actual BadMapErrors from the Map.get calls 

Nahh, let's not worry about it. The key thing here is printing out any errors we got back from the API, which we now do on line 80.

This is a pretty good application of Elixir's "let it crash" philosophy. If we have already confirmed that the response is a success message and the data is _still_ not in the format we expect, then something extremely unexpected has happened (such as start.gg changing their API) and there's really nothing we can do about it anyway. Might as well let it crash. :man_shrugging: